app/main.py

```diff
@@ -1,8 +1,23 @@
 class Person:
-    # write your code here
-    pass
 
+    people = {}
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
+
+
+def create_person_list(persons: list[dict]) -> list[Person]:
+
+    persons_instances = [Person(person["name"], person["age"])
+                         for person in persons]
+
+    for i in range(len(persons_instances)):
+        if "wife" in persons[i] and persons[i]["wife"] is not None:
```

Comment:
You don’t need indexes here all over the loop, use value from your constructor: `Person.people[self.name] `
As a result, you will not need `range(len(…))`